src/Chainweb/Miner/Miners.hs

```diff
@@ -96,6 +102,22 @@ localTest lf v m cdb gen miners = runForever lf "Chainweb.Miner.Miners.localTest
     work :: BlockHeader -> IO BlockHeader
     work bh = MWC.geometric1 t gen >>= threadDelay >> pure bh
 
+-- | A miner that grabs new blocks from mempool and discards them. Mempool
+-- pruning happens during new-block time, so we need to ask for a new block
+-- regularly to prune mempool.
+mempoolNoopMiner
+    :: LogFunction
+    -> HashMap ChainId (ChainResources logger)
+    -> IO ()
+mempoolNoopMiner lf chainRes =
+    runForever lf "Chainweb.Miner.Miners.mempoolNoopMiner" loop
```

Comment:
Only the `_chainResMempool` is required. Passing the whole structure my keep references to more resources alive than actually needed.

Generally, the `*Resources` types are supposed to only be used during initialization. The idea is that on initialization each component picks from the resources only what it actually needs.